ios/Classes/MapboxMapController.swift

```diff
@@ -119,6 +119,13 @@ class MapboxMapController: NSObject, FlutterPlatformView, MGLMapViewDelegate, Ma
         mapView.minimumZoomLevel = min
         mapView.maximumZoomLevel = max
     }
+    func setStyleString(style: String) {
```

Comment:
Would you be able to expand the functionality of this method to match Android equivalent? For the styleString setter we also check if the provided string starts with "{" and ends with "}", if true we set the string as the map styleJSON. Android code is https://github.com/tobrun/flutter-mapbox-gl/blob/master/android/src/main/java/com/mapbox/mapboxgl/MapboxMapController.java#L234

Reply:
Unlike the Android SDK, the iOS SDK does not have a Style Builder. I can add the checks and load the style from URL if it is not empty and does not start with a bracket.

Some remarks:

- The style string is never null (or nil) at this point (can be empty). In the Convert class the string is unwrapped and when the styleString is null the setStyleString is not called.
- When logging in iOS (with NSLog) nothing is shown in the flutter console. Using the new unified logging system (os_log which required iOS 10.0 or higher) also does not "work". I'll add some log lines for when it is (hopefully) supported in the future. Also see: https://github.com/flutter/flutter/issues/13204

Reply:
it seems you are right that it isn't supported right now. Though I feel it should, I created a feature request for it upstream https://github.com/mapbox/mapbox-gl-native/issues/14137. That said, I think this PR is ready to be merged!

Reply:
If all you have is the JSON and no file on disk, a workaround would be to save the JSON to a temporary file and set the style URL to that file’s URL.